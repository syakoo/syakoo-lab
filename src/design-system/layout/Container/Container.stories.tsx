import type { Meta } from "@storybook/react";
import type React from "react";

import { tokens } from "@/design-system/tokens";

import { Container } from ".";

const meta = {
  component: Container,
  tags: ["autodocs"],
  parameters: { testLevel: "snapshot" },
} satisfies Meta<typeof Container>;

export default meta;

const SampleInnerContent = ({ children }: { children: React.ReactNode }) => (
  <div
    style={{
      borderRadius: tokens.radii[100],
      background: tokens.colors.palette.gray[100],
      padding: "50px",
    }}
  >
    {children}
  </div>
);

export const Sample = () => {
  return (
    <div style={{ display: "flex", flexDirection: "column", gap: "50px" }}>
      <div>
        <Container>
          <SampleInnerContent>size: 100 (default)</SampleInnerContent>
        </Container>
      </div>
      <div>
        <Container size="50">
          <SampleInnerContent>size: 50</SampleInnerContent>
        </Container>
      </div>
    </div>
  );
};
