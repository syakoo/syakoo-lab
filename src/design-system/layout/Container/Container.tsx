import React from "react";
import type { RecipeVariants } from "@vanilla-extract/recipes";
import { containerStyle } from "./Container.css";
import { PolymorphicComponent } from "@/utils/polymorphic-component";

type ContainerProps = {
  as?: React.ElementType;
  children: React.ReactNode;
} & RecipeVariants<typeof containerStyle>;

/**
 * 子要素を格納するサイズを決定するレイアウトコンテナコンポーネント
 */
export const Container: React.FC<ContainerProps> = ({
  as = "div",
  children,
  size = "100",
  ...otherVariantProps
}) => (
  <PolymorphicComponent
    as={as}
    // NOTE: polymorphic component なのでヨシ
    // eslint-disable-next-line react/forbid-component-props
    className={containerStyle({ size, ...otherVariantProps })}
  >
    {children}
  </PolymorphicComponent>
);
