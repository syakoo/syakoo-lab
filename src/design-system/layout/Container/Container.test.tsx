import { composeStories } from "@storybook/react";
import { act, render } from "@testing-library/react";

import * as stories from "./Container.stories";

const { ...otherStories } = composeStories(stories);

describe("Container", () => {
  const testCases = Object.values(otherStories).map(
    (Story) => [Story.storyName, Story] as const,
  );
  test.each(testCases)("renders %s", async (_, Story) => {
    const tree = render(<Story />);
    await act(async () => {
      if (Story.play) {
        await Story.play({ canvasElement: tree.container });
      }
    });

    expect(tree.baseElement).toMatchSnapshot();
  });
});
