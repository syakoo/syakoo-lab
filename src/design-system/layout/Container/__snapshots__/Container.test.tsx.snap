// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container スナップショット Sample 1`] = `
<body>
  <div>
    <div
      style="display: flex; flex-direction: column; gap: 50px;"
    >
      <div>
        <div
          class="Container_containerStyle__1b3ljla3 Container_containerStyle_size_100__1b3ljla5 Container_containerStyle_variants_size_100__1b3ljla1"
        >
          <div
            style="border-radius: var(--radius-100__13jk9q11g); padding: 50px;"
          >
            size: 100 (default)
          </div>
        </div>
      </div>
      <div>
        <div
          class="Container_containerStyle__1b3ljla3 Container_containerStyle_size_50__1b3ljla4 Container_containerStyle_variants_size_50__1b3ljla0"
        >
          <div
            style="border-radius: var(--radius-100__13jk9q11g); padding: 50px;"
          >
            size: 50
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
